### PR TITLE
feat : add page detail complaint from span

### PR DESCRIPTION
**Pull request by marsellavaleria19**
## Related
https://app.clickup.com/t/860rqta50

## Changes
- [add component dialog add complaint & show data complaint on component complaint](https://github.com/jabardigitalservice/super-app-cms/commit/d306435ee286aec02a1829d754ee7882a3231f62)
- [change width on component complaint detail](https://github.com/jabardigitalservice/super-app-cms/commit/8d2720bf730d980027b15ec6ceb4b1739ba9ed3a)
- [add page detail complaint from span](https://github.com/jabardigitalservice/super-app-cms/commit/b923721dbbf46fe837f7f72e2f8da84bdae43e2f)

## Preview
image

## Evidence
title: add page detail complaint from span
project: Jabar Super Apps
participants: @marsellavaleria19 @doohanas @rayfajars

### CODE REVIEW

File: components/Aduan/Dialog/AddComplaint/Form/InformationComplaint/index.vue [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+          <ValidationProvider v-slot="{ errors }" name="ID SP4N Lapor" rules="required|numeric">
+            <BaseInputText
+              v-model="dataInformationComplaint.sp4n_id.value"
+              placeholder="Masukkan ID SP4N Lapor"
+              :error-message="(dataInformationComplaint.sp4n_id.isInputDirty || isSubmit) ? errors[0] : ''"
+              label="ID SP4N Lapor"
+              maxlength="10"
+            />
+          </ValidationProvider>
+        </div>
```
Comment (doohanas) [inline components/Aduan/Dialog/AddComplaint/Form/InformationComplaint/index.vue:18]: ### Suggestion

Hello @marsellavaleria19, Instead you add `div` as wrapper, we can add tag props in ValidationProvider, ex:

```Vue
  <ValidationProvider class="mb-4" v-slot="{ errors }" name="ID SP4N Lapor" rules="required|numeric" tag="div">
     <BaseInputText
          v-model="dataInformationComplaint.sp4n_id.value"
          placeholder="Masukkan ID SP4N Lapor"
          :error-message="(dataInformationComplaint.sp4n_id.isInputDirty || isSubmit) ? errors[0] : ''"
          label="ID SP4N Lapor"
          maxlength="10"
        />
  </ValidationProvider>
```

source: https://vee-validate.logaretm.com/v2/guide/components/validation-provider.html#rendering

File: components/Aduan/Dialog/AddComplaint/Form/InformationComplaint/index.vue [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            <label class="text-gray-800 text-[15px] mb-1">Tanggal Laporan Masuk</label><br>
+            <date-picker
+              v-model="dataInformationComplaint.span_created_at.value"
+              format="DD/MM/YYYY"
+              :class="{'mx-datepicker--error': (dataInformationComplaint.span_created_at.isInputDirty || isSubmit) ? errors[0] : ''}"
+              placeholder="Pilih Tanggal Laporan Masuk"
+            /><br>
+            <small class="text-red-600">{{ (dataInformationComplaint.span_created_at.isInputDirty || isSubmit) ? errors[0] : '' }}</small>
+          </ValidationProvider>
+        </div>
```
Comment (doohanas) [inline components/Aduan/Dialog/AddComplaint/Form/InformationComplaint/index.vue:30]: Same here too

File: components/Aduan/Dialog/AddComplaint/Form/InformationComplaint/index.vue [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+          <ValidationProvider v-slot="{ errors }" name="Judul Aduan" rules="required">
+            <jds-input-text
+              v-model="dataInformationComplaint.title.value"
+              name="Judul Aduan"
+              label="Judul Aduan"
+              placeholder="Masukkan Judul Aduan"
+              :error-message="(dataInformationComplaint.title.isInputDirty || isSubmit) ? errors[0] : ''"
+            />
+          </ValidationProvider>
+        </div>
```
Comment (doohanas) [inline components/Aduan/Dialog/AddComplaint/Form/InformationComplaint/index.vue:52]: same here too

File: components/Aduan/Dialog/AddComplaint/Form/InformationComplaint/index.vue [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+              label="Isi Laporan"
+              name="Isi Laporan"
+              maxlength="255"
+              :error-message="(dataInformationComplaint.description.isInputDirty || isSubmit) ? errors[0] : ''"
+            />
+          </ValidationProvider>
+          <p class="text-sm text-gray-600">
+            Tersisa {{ 255-dataInformationComplaint.description.value.length }} karakter
+          </p>
+        </div>
```
Comment (doohanas) [inline components/Aduan/Dialog/AddComplaint/Form/InformationComplaint/index.vue:67]: same here too

File: components/Aduan/Dialog/AddComplaint/Form/LocationComplaint/index.vue [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+            <jds-select
+              v-model="dataLocationComplaint.city_id.value"
+              name="Kota/Kabupaten"
+              label="Kota/Kabupaten"
+              placeholder="Pilih Kota/Kabupaten"
+              :error-message="(dataLocationComplaint.city_id.isChooseDirty || isSubmit) ? errors[0] : ''"
+              :options="listCity"
+            />
+          </ValidationProvider>
+        </div>
```
Comment (doohanas) [inline components/Aduan/Dialog/AddComplaint/Form/LocationComplaint/index.vue:19]: Same here too

File: components/Aduan/Dialog/AddComplaint/Form/LocationComplaint/index.vue [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+            <jds-select
+              v-model="dataLocationComplaint.district_id.value"
+              name="Kecamatan"
+              label="Kecamatan"
+              placeholder="Pilih Kecamatan"
+              :error-message="(dataLocationComplaint.district_id.isChooseDirty || isSubmit) ? errors[0] : ''"
+              :options="listDistrict"
+            />
+          </ValidationProvider>
+        </div>
```
Comment (doohanas) [inline components/Aduan/Dialog/AddComplaint/Form/LocationComplaint/index.vue:31]: Same here too

File: components/Aduan/Dialog/AddComplaint/Form/LocationComplaint/index.vue [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+            <jds-select
+              v-model="dataLocationComplaint.subdistrict_id.value"
+              name="Kelurahan"
+              label="Kelurahan"
+              placeholder="Pilih Kelurahan"
+              :error-message="(dataLocationComplaint.subdistrict_id.isChooseDirty || isSubmit) ? errors[0] : ''"
+              :options="listVillage"
+            />
+          </ValidationProvider>
+        </div>
```
Comment (doohanas) [inline components/Aduan/Dialog/AddComplaint/Form/LocationComplaint/index.vue:43]: same here too

File: components/Aduan/Dialog/AddComplaint/Form/LocationComplaint/index.vue [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+              label="Detail Lokasi"
+              name="Detail Lokasi"
+              maxlength="255"
+              :error-message="(isInputDirty || isSubmit) ? errors[0] : ''"
+            />
+          </ValidationProvider>
+          <p class="text-sm text-gray-600">
+            Tersisa {{ 255-addressDetail.length }} karakter
+          </p>
+        </div>
```
Comment (doohanas) [inline components/Aduan/Dialog/AddComplaint/Form/LocationComplaint/index.vue:58]: same here too

File: components/Aduan/Dialog/AddComplaint/Form/LocationComplaint/index.vue [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+    },
+    listDistrict () {
+      return this.listDataDistrict.map((item) => {
+        return { value: item.id, label: item.name }
+      })
+    },
+    listVillage () {
+      return this.listDataVillage.map((item) => {
+        return { value: item.id, label: item.name }
+      })
```
Comment (doohanas) [inline components/Aduan/Dialog/AddComplaint/Form/LocationComplaint/index.vue:135]: ### Suggestion

#### [Readability] Concern

- Hello @marsellavaleria19, we can use meaningful variable names: The variable name `item` is generic and does not provide much context. Consider using a more descriptive name that reflects the purpose of the variable.

example: 
```js
listCity() {
  return this.listDataCity.map((city) => {
    return { value: city.id, label: city.name };
  });
}
```

**Review by rayfajars — APPROVED** [review]

**Pull request merged**